odbc/src/tests.rs

```diff
@@ -287,3 +291,163 @@ fn invalid_alloc() {
         );
     }
 }
+
```

Comment:
can we put all of these new tests into an inline `get_diag_rec` module to isolate them from the other tests in this file?